doc/upstream_release_process.md

```diff
@@ -172,13 +172,19 @@ Questions to ask / answer:
 These checks should be automatable.
 
 ## Upload to ubuntu/devel
+
 ```bash
 $ git fetch upstream
 $ git checkout upstream/ubuntu/devel -B ubuntu/devel
-$ new_upstream_snapshot.py  # add the Release LP: #
-$ <run whatever commands new-upstream-snapshot tells you to run next to finish the release>
+
+# add the Release LP: #
+$ new_upstream_snapshot.py -c <version>
```

Comment:
I think we opted this release to just cut whatever is in tip of upstream/main `-c upstream/main` or specific commit from `upstream/main` because main should typically be releasable to devel any time we want because our CI/Jenkins review prior to release and general PR review keeps things generally safe for devel release.

We should add content if we don't already have it, that describes the hotfix process: Here are some thoughts for discussion.
### PR 1 
* git checkout 23.1 -B 23.1.x; git push push to upstream/23.1.x   # create hotfix branch
* cherry-pick specific commits from `upstream/main` into that branch
* single supplemental commit on 23.1.x branch which adds a new 23.1.1 section to ChangeLog with the now included commit summaries in 23.1.1. Example https://github.com/canonical/cloud-init/releases/tag/23.1.1
* git push YOUR_REMOTE 23.1.x
* setup PR against upstream/23.1.x for peer review
### PR 2: Once PR1 lands
* push annotated tag 23.1.1 to upstream
* git checkout upstream/main -B main
* git cherry-pick topmost commit from 23.1.1
* Create a PR against upstream/main for inclusion of cloudinit/version.py ++ ChangeLog
### PR 3: Once PR2 lands
* git checkout upstream/ubuntu/devel -B ubuntu/devel
* new_upstream_snapshot.py -c upstream/main (to pull in latest and greatest releasable content from tip of main (and ensure our versioning in lunar stays ahead (dpkg --compare-versions lunar_pkg_version gt kinetic_pkg_version) of any SRUs of 23.1.1 we push to downstream.